### PR TITLE
Updated README and added comments around modules.

### PR DESCRIPTION
**Pull request by bendory**

### CODE REVIEW

File: go/prepare_workspace.inc [hunk 1]
```diff
@@ -13,10 +13,17 @@
 # limitations under the License.
 
 prepare_workspace() {
+	# Use Go modules (https://golang.org/ref/mod) if defined. This is the
+	# prefered workspace setup; it has been available since Go 1.11 and is the
+	# default behavior as of 1.16.
```
Comment (ronanww) [inline go/prepare_workspace.inc:18]: indentation seems off. It looks like here it was done with `tab` but in the rest of the file the indentation was done with `spaces`

File: go/prepare_workspace.inc [hunk 2]
```diff
@@ -13,10 +13,17 @@
...
     if [[ -e "go.mod" ]]; then
         return 0
     fi
 
+	# Workspace setup without modules lives on for legacy purposes; it is no
+	# longer a supported behavior. Users should upgrade to use Go modules.
+	# See https://golang.org/ref/mod
+
```
Comment (ronanww) [inline go/prepare_workspace.inc:26]: same comment as above ^

File: go/prepare_workspace.inc (hunk 1 above)
Comment (bendory) [reply go/prepare_workspace.inc]: thanks, fixed

File: go/prepare_workspace.inc (hunk 2 above)
Comment (bendory) [reply go/prepare_workspace.inc]: thanks, fixed

**Review by ronanww — APPROVED** [review]
👍

**Pull request merged**